### PR TITLE
issue-528-handling-user-errors-for-OpenSearch

### PR DESCRIPTION
**Pull request by OleksiienkoMykyta**

### CODE REVIEW

**Review by worryg0d — APPROVED** [review]

**Review by tengu-alt — APPROVED** [review]

File: controllers/clusterresources/helpers.go [hunk 1]
```diff
@@ -67,6 +70,31 @@ func areEncryptionKeyStatusesEqual(a, b *v1beta1.AWSEncryptionKeyStatus) bool {
...
+		return nil, err
+	}
+
+	err = json.Unmarshal(b, &users)
+	if err != nil {
+		return nil, err
+	}
+
+	return users, nil
+}
```
Comment (ribaraka) [inline controllers/clusterresources/helpers.go:96]: This function seems redundant. All this code may be moved to `api.FetchUsers(clusterID, app)`

File: controllers/clusterresources/opensearchuser_controller.go [hunk 2]
```diff
@@ -205,19 +205,31 @@ func (r *OpenSearchUserReconciler) createUser(
...
 	}
 
-	err = r.API.CreateUser(user.ToInstaAPI(username, password), clusterID, models.OpenSearchAppKind)
+	exists, err := CheckIfUserExists(username, clusterID, models.OpenSearchAppKind, r.API)
 	if err != nil {
-		logger.Error(err, "Cannot create OpenSearch user on Instaclustr",
-			"username", username,
-			"cluster ID", clusterID,
-		)
+		logger.Error(err, "Cannot check if user exists ")
```
Comment (ribaraka) [inline controllers/clusterresources/opensearchuser_controller.go:210]: Have you tested a case when a cluster was deleted? What error code do we get?

**Review by ribaraka — CHANGES_REQUESTED** [review]

File: controllers/clusterresources/helpers.go (hunk 1 above)
Comment (OleksiienkoMykyta) [reply controllers/clusterresources/helpers.go]: fixed

File: controllers/clusterresources/opensearchuser_controller.go (hunk 2 above)
Comment (OleksiienkoMykyta) [reply controllers/clusterresources/opensearchuser_controller.go:210]: I have tested it, we getting 404 not found

Comment (OleksiienkoMykyta) [reply controllers/clusterresources/opensearchuser_controller.go:210]: I have tested it, we getting 404 not found

Comment (ribaraka) [reply controllers/clusterresources/opensearchuser_controller.go:210]: The error is returned at the endpoint when we fetch all users in a cluster. 
For now, let's just handle it as a regular error.

**Review by ribaraka — APPROVED** [review]

**Review by RostislavPorohnya — APPROVED** [review]

**Review by taaraora — APPROVED** [review]

**Pull request merged**